pkg/controller/jobframework/reconciler.go

```diff
@@ -473,16 +475,35 @@ func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, o
 	return match, nil
 }
 
+func FindMatchingWorkloads(ctx context.Context, c client.Client, job GenericJob, object client.Object) (match *kueue.Workload, toDelete []*kueue.Workload, workloads *kueue.WorkloadList, err error) {
```

Comment:
```suggestion
func findMatchingWorkloads(ctx context.Context, c client.Client, job GenericJob, object client.Object) (match *kueue.Workload, toDelete []*kueue.Workload, workloads *kueue.WorkloadList, err error) {
```

Reply:
I'm using this in the pod controller for the case of a single pod:

https://github.com/kubernetes-sigs/kueue/pull/1319/files#diff-dfb49586a8522fa91d733051fd3b7e4b3ff174907898cf249ab16e2620976a5dR432